### ISSUE TITLE
Discussion | GNU/Linux Mobile

### ISSUE THREAD

**Issue by ggg27** [FreeTubeApp/FreeTube#337]
**Description**: Supporting GNU/Linux for Mobile Devices
(i.e. [Ubuntu Touch](https://www.wikipedia.org/wiki/Ubuntu_Touch), [Purism tablet/phone](https://www.wikipedia.org/wiki/Librem#Librem_11_tablet), [PinePhone](https://www.wikipedia.org/wiki/Pine_Microsystems#Smartphone), [PineTab](https://www.pine64.org/pinetab/), e.c.t)
**How?**: In theory most of it should still work, the key piece is verifying the UI works on a much smaller touch screen (i.e [adaptive UI](https://www.wikipedia.org/wiki/Adaptive_user_interface)).

__________________

Want to support this issue? Post a [bounty on it](https://www.bountysource.com/issues/78761559-discussion-gnu-linux-mobile)! We accept bounties via [BountySource](https://www.wikipedia.org/wiki/Bountysource).

**Comment by PrestonN** [comment]
This is something I've been thinking about for quite a while now.  Personally I'd love to see it happen.

At this moment, FreeTube already provides ARM builds which should theoretically work on the Librem 5 as-is with no further modifications.  Having said that, there's a good chance that it may not be a great experience with how FreeTube is at it's current state.

Most of the UI should resize properly to the small display size, though there are definitely some tweaks to be make in order for it to be a better experience.  It also wouldn't incorporate any sort of touch gestures (such as sliding from the side to bring up the side menu).

There's also the potential problem on performance.  With FreeTube being an Electron app, I'm not sure how well performance will be on a device like the Librem 5, especially when resources and battery life are at a premium.

I will likely be purchasing a Librem 5 whenever it comes out, so I will be able to do some actual tests with it.  I will attempt to get a VM up and running with their OS to get an idea on how well it runs though.

So yes, it should work, but there will likely be a lot of changes that need to be made in order for it to work well on the device.  I'm prepared to make it work as well as I can when the device is released.

**Comment by PrestonN** [comment]
I'll have more news about this in the future, though I have successfully been able to get FreeTube running on a Pinephone.  I don't have too much more to say on it though as there are still quite a few issues with it and won't be a priority for now, but some progress has been made at least.

**Comment by okias** [comment]
If needed, I can give a shot on Librem 5 devkit (== final phone at least for screen & handling).

**Comment by PrestonN** [comment]
I already own a PInephone that I can use for testing if needed.  Most issues that I'll have on the Pinephone will likely be similar on the Librem 5.  The app technically already works on the device but it's very unusable in it's current state and needs some dedicated tweaking to make it work.

**Comment by truthinhoney** [comment]
Tubefeeder is a project that is _similar_ to FreeTube, and it's specifically made for GNU/Linux mobile devices.

https://github.com/Tubefeeder/Tubefeeder

https://github.com/Tubefeeder/tubefeeder-extractor

From my understanding, Tubefeeder is currently being rewritten at the time that I'm writing this comment.

There is a GNOME media player called Clapper that can do hardware acceleration video playback on the PinePhone. 

Clapper can be used with Tubefeeder :100:

**Comment by ckyle6300** [comment]
I have the pinephone and Freetube works well on it for the most part. The only 2 problems are that you can not get to your subscriptions easily. The other is that it appears whenever videos are fetched by the invidious api, the video size does not stay a standard size.

**Comment by peepopoggers** [comment]
@ckyle6300 open issue threads on this repository for these issues you describe

**Comment by julianfairfax** [comment]
I tried FreeTube on my PinePhone running Mobian and it worked well. Maybe this issue should be closed?

**Comment by tootbrute** [comment]
> I tried FreeTube on my PinePhone running Mobian and it worked well. Maybe this issue should be closed?

I am running it on Mobian+Phosh on Pinephone Beta but it doesn't work for me.

- App runs.
- Resized ok for the screen except in playlists.
- Cursor doesn't always go to the right place, need to click where the cursor goes.
- Search works but playing videos doesn't work at all.
- Hard to tell what tab you are in because the bottom buttons don't change color.
- App is very slow. Uses 45% of CPU and 300 MB of RAM.

**Comment by ChunkyProgrammer** [comment]
As FreeTube does run on Linux mobile I'm going to go ahead and close this issue. If you notice any issues on your phone, feel free to open separate issues, thanks 😄. The next version of FreeTube should have better performance but it's still an electron app.
